Data/Row.hs

```diff
@@ -51,7 +51,7 @@ module Data.Row
   )
 where
 
-import Data.Row.Variants
+import Data.Row.Variants hiding (restrict, split)
```

Comment:
The model that I was going for was that `Data.Row` would only export names that are:

- Shared within row-types (e.g. `Row`)
- Not in the standard prelude or from typical libraries (e.g. not `map`)
- Unique in this library (e.g. `trial`)

The `restrict` function used to fall into the last category, which is why it was being exported, but if `Data.Row.Variants` is now going to be exporting a `restrict` function, I'd rather we remove it from the export list of `Data.Row`.  I know this makes this a breaking change, but it feels more consistent this way.